docs/dev/adding-new-platform.md

```diff
@@ -0,0 +1,198 @@
+
+Once the platform can be launched and tested, system features must be implemented. The sections below are roughly independent:
+
+* General requirements:
+    * Replace the installer terraform destroy with one that doesn't rely on terraform state
+    * Ensure the installer IPI support is consistent with other platform features (private config, etc)
+        * Not all platforms will support all features, so IPI is taken to be a spectrum of support
+    * Enable a CI job that verifies the e2e suite for the given platform runs successfully
+* Requirements for dynamic storage and dynamic load balancing
+    * Ensure the cloud provider is enabled in Kubernetes for your platform (this is required for `supported UPI with cloud provider`)
+    * Enable cluster-storage-operator to set the correct default storage class
+* Requirements for dynamic compute:
+    * Enable the cloud credential operator for the platform to subdivide access for individual operators
+    * Enable dynamic compute (MachineSets) by adding a cloud actuator for that platform
+* Requirements for dynamic ingress and images:
+    * Enable cluster-image-registry-operator to provision a storage bucket (if your platform supports object storage)
+    * Enable cluster-ingress-operator to provision the wildcard domain names
+
+At this point the platform is said to be a `supported IPI with Dynamic Compute` if the platform supports
+MachineSets, or `supported IPI without Dynamic Compute` if it does not.
+
+
+OpenShift Architectural Design
+------------------------------
+
+OpenShift 4 combines Kubernetes, fully-managed host lifecycle, dynamic infrastructure management, and a comprehensive set of fully-automated platform extensions that can be upgraded and managed uniformly.  The foundation of the platform is Red Hat CoreOS, an immutable operating system based on RHEL that is capable of acting as a fully-integrated part of the cluster. The Kubernetes control plane is hosted on the cluster, along with a number of other fundamental extensions like cluster networking, ingress, storage, and application development tooling. Each of those extensions is fully managed on-cluster via a cluster operator that reacts to top level global configuration APIs and can automatically reconfigure the affected components. The Operator Lifecycle Manager (OLM) allows additional ecosystem operators to be installed, upgraded and managed. All of these components - from the operating system kernel to the web console - are part of a unified update lifecycle under the top level Cluster Version Operator which offers worry free rolling updates of the entire infrastructure.
+
+### Core configuration
+
+An OpenShift cluster programs the infrastructure it runs on to provide operational simplicity. For every platform, the minimum requirements are:
+
+1. The control plane nodes:
+   1. Run RH CoreOS, allowing in-place updates
+   2. Are fronted by a load balancer that allows raw TCP connections to port 6443 and exposes port 443
```

Comment:
To be IaaS neutral, wouldn't it be possible to use Keepalive (within Kube since RHCOS is immutable)? It could be used either as LB or failover handler. Not using AWS doesn't automatically means having a hardware LB in front of a cluster.